pkg/controller/actuator_test.go

```diff
@@ -312,6 +314,15 @@ var _ = Describe("actuator test", func() {
 			Expect(a.Reconcile(ctx, logger, ext)).To(Succeed())
 
 			// assert
+			mr := &v1alpha1.ManagedResource{}
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: ResourceNameSeed, Namespace: shootNamespace1}, mr)).To(Succeed())
+			secret := &corev1.Secret{}
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: mr.Spec.SecretRefs[0].Name, Namespace: shootNamespace1}, secret)).To(Succeed())
+			Expect(secret.Data["seed"]).To(ContainSubstring("1.2.3.4"))
+			Expect(secret.Data["seed"]).To(ContainSubstring(istioNamespace1))
+			Expect(secret.Data["seed"]).To(ContainSubstring("acl-vpn-" + shootNamespace1))
+
+			// check legacy envoyfilter
```

Comment:
Can you also add a test case for ensuring the shoot-specific part of the legacy filter is removed if the new filter is already present?